src/java.base/unix/classes/java/io/UnixFileSystem.java

```diff
@@ -101,15 +101,23 @@ public int prefixLength(String pathname) {
         return pathname.startsWith("/") ? 1 : 0;
     }
 
+    // Remove trailing '/' if present and there are at least two characters
+    private static String trimSeparator(String s) {
+        int len = s.length();
+        if (len > 1 && s.charAt(len - 1) == '/')
+            return s.substring(0, len - 1);
+        return s;
```

Comment:
If there are multiple trailing `/`, is it sufficient to remove only a single `/`?
Does this have the same result as `normalize(pathname)`?

Reply:
The main calling context is
```
            if (parent.path.isEmpty()) {
                this.path = FS.resolve(FS.getDefaultParent(),
                                       FS.normalize(child));
            } else {
                this.path = FS.resolve(parent.path,
                                       FS.normalize(child));
            }
```
where `parent` is non-`null` and already normalized. Thus both components should already be normalized and multiple trailing `/` impossible. I _think_ that it should give the same result as `normalize(pathname)`.